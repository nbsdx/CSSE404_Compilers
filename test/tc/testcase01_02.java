// Wrong return type
class A1 {
    public static void main(String[] a2) {
	System.out.println(1);
    }
}

class double {}

class A3 {
    public double a4(int a5) {
	return a5*1;
    }
}

class A6 extends A3 {
    public boolean a7() {
	return this.a5(4)==4;
    }
    
    public double a4(int a5) {
	return this.a7(); //Returning a boolean instead of a double
    }
}
