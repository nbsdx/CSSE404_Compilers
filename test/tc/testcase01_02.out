Actual return type int of method a4 does not match declared type double
No method named a5 found for class A6
The operand types, null and int, are not compatible for equality comparison
Actual return type boolean of method a4 does not match declared type double
